[Build] Enable Non-Transitive Resources

ParaskP7 opened the pull request:
This PR is a prerequisite for the `Gradle 8.1.1 & AGP 8.0.2 Upgrade for WPAndroid, WCAndroid & Related Libs` project.

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

-----

This PR enables non-transitive resources (`android.nonTransitiveRClass`) for the project.

FYI: This behavior becomes the default in AGP `8.0` and higher. As such, this becomes a prerequisite for the AGP `8.0.2` upgrade, that is of course, unless `android.nonTransitiveRClass` is explicitly set to `false`.

-----

Dependency Versions Refactor List:

1. [Extract automattic publish to s3 version to settings build gradle](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/68/commits/5ee4b6fa357a01dccdd420c6edd2641638978092)

-----

## To test:

1. Verify that all the CI checks are successful.
3. Smoke test the app.

ParaskP7 committed:
Build: Move android related gradle properties to its own group

ParaskP7 committed:
Refactor: Optimize imports on res for org wordpress android mp R

ParaskP7 committed:
Refactor: Optimize imports on res for org wordpress android mp api R

ParaskP7 committed:
Refactor: Optimize imports on res for org wordpress android sampleapp R

ParaskP7 committed:
Build: Automatically migrate to non-transitive r classes

The changes in this commit was automatically applied on triggering the
"Refactor" -> "Migration to Non-Transitive R Classes..." migration
process via AS.

ParaskP7 committed:
Refactor: Optimize imports on res for org wordpress android mp api R

ParaskP7 committed:
Build: Enable non transitive resources for the project

ParaskP7 committed:
Deps: Extract automattic publish to s3 version to settings build gradle

oguzkocer commented on gradle.properties:
```diff
@@ -1,3 +1,10 @@
+# Project-wide Gradle settings.
+
+org.gradle.jvmargs=-Xmx1536m
```
When we are making these changes, it'd be great if we can take the opportunity to increase this value. We used to have it set low for most projects because CircleCI machines only had 2GB of memory. However, our current Android CIs have significantly more, so I usually set these to 6G whenever I get the chance.

I realize that this is just moving the current value, so I'll leave it to you to whether to include my suggestion in this PR or not.

ParaskP7 replied:
Great suggestion @oguzkocer , keep them coming! 🙇 ❤️ 🚀 

Done: a477e091424f145caf96d7e8bdc351a3b89ccb13

ParaskP7 committed:
Build: Update org gradle jvmargs gradle property to xmx 6g

This gradle property used to have a low value for most projects because
'CircleCI' machines only had '2GB' of memory. 'Buildkite' machines have
significantly more, so setting this gradle property to '6G' of memory is
the new way.

ParaskP7 commented:
Thank you for reviewing and testing this @oguzkocer , you rock! 🙇 ❤️ 🚀
